packages/runtime/container-runtime/src/opLifecycle/README.md

```diff
@@ -155,3 +171,98 @@ Notice that the sequence numbers don’t matter here, as all ops will be based o
 Additionally, as compression preserves the original uncompressed batch layout in terms of the number of ops by using empty ops to reserve the sequence numbers, this ensures that the clients will always receive the exact count of ops to rebuild the uncompressed batch sequentially.
 
 On the receiving end, the client will accumulate chunks 1 and 2 and keep them in memory. When chunk 3 is received, the original large, decompressed op will be rebuilt, and the runtime will then process the batch as if it is a compressed batch.
+
+## How grouped batching works
+
+Given the following baseline batch:
+
+```
++---------------+---------------+---------------+---------------+---------------+
+| Op 1          | Op 2          | Op 3          | Op 4          | Op 5          |
+| Contents: "a" | Contents: "b" | Contents: "c" | Contents: "d" | Contents: "e" |
++---------------+---------------+---------------+---------------+---------------+
+```
+
+Compressed batch:
+
+```
++--------------------+-----------------+-----------------+-----------------+-----------------+
+| Op 1               | Op 2            | Op 3            | Op 4            | Op 5            |
+| Contents: "abcde"  | Contents: empty | Contents: empty | Contents: empty | Contents: empty |
+| Compression: 'lz4' |                 |                 |                 |                 |
++--------------------+-----------------+-----------------+-----------------+-----------------+
+```
+
+Grouped batch:
+
+```
++---------------------------------------------------------------------------------------------------------------------------------+
+| Op 1                   Contents: +--------------------+-----------------+-----------------+-----------------+-----------------+ |
+| SeqNum: 1                        | Op 1               | Op 2            | Op 3            | Op 4            | Op 5            | |
+| Type: "groupedBatch"             | Contents: "abcde"  | Contents: empty | Contents: empty | Contents: empty | Contents: empty | |
+|                                  | Compression: 'lz4' |                 |                 |                 |                 | |
+|                                  +--------------------+-----------------+-----------------+-----------------+-----------------+ |
++---------------------------------------------------------------------------------------------------------------------------------+
+```
+
+Chunked batch:
+
+```
++---------------------------------------------------------------------------------------------------------------------------------+
```

Comment:
I really want to get rid of empty ops :) So I think we should say somewhere that it's point in time and point to a task that tracks that work. I think we need to do it before we enable it in production.